src/sources/metadatasourcetaglib.h

```diff
@@ -7,16 +7,11 @@ namespace mixxx {
 // Universal default implementation of IMetadataSource using TagLib.
 class MetadataSourceTagLib : public MetadataSource {
   public:
-    explicit MetadataSourceTagLib(
-            const QString& fileName)
-            : m_fileName(fileName),
-              m_fileType(taglib::getFileTypeFromFileName(fileName)) {
-    }
     MetadataSourceTagLib(
             const QString& fileName,
-            taglib::FileType fileType)
+            const QString& fileType)
             : m_fileName(fileName),
-              m_fileType(fileType) {
+              m_fileType(taglib::stringToEnumFileType(fileType)) {
```

Comment:
whats the advantage of taking a string here instead of shifting the conversion duty to the caller? This smells [stringly-typed](https://wiki.c2.com/?StringlyTyped)

Reply:
fyi, two of the other three comments were redundant if this would avoid strings better.

Reply:
We have the duality of a string typed fileType used in Soundsource classes outside of Taglib and the enum inside Taglib. We still need the "fileType" as string, because it holds the file extension in case the type has not been normalized.  This is a left over from the legacy when Mixxx uses just the file extension.

I have moved the conversion here, to hide all implementation details of MetadataSourceTagLib and keep the taglib includes in its cpp file. 
Else it would be here: 
https://github.com/mixxxdj/mixxx/blob/d1c09b5c5006ef674cb4ab8eaa4fb617200c79d7/src/sources/soundsource.cpp#L84-L88